### PR TITLE
fix: handle missing detach events for restored bfcache targets

### PR DESCRIPTION
**Pull request by OrKoN**
Currently, if we navigate from a page with a worker in an OOPIF to another, we don't get a target detached event for the worker. The session and the target, therefore, continues to be registered as "attached" by Puppeteer. This leads to an issue when on going back to the same page the worker target is double attached. This behaviour is probably caused by the fact that the page is put into a bfcache instead of being actually destroyed. Probably, the correct behaviour from the CDP side would be to keep the client attached to the cached session instead of creating new sessions. Currently, this is not the case so I filed https://crbug.com/1485193.

Closes #10946 #10966

### CODE REVIEW

File: packages/puppeteer-core/src/cdp/ChromeTargetManager.ts [hunk 1]
```diff
@@ -358,10 +359,7 @@ export class ChromeTargetManager
     // `this.#connection.isAutoAttached(targetInfo.targetId)`. In the future, we
     // should determine if a target is auto-attached or not with the help of
     // CDP.
-    if (
-      targetInfo.type === 'service_worker' &&
-      this.#connection.isAutoAttached(targetInfo.targetId)
-    ) {
+    if (targetInfo.type === 'service_worker') {
```
Comment (OrKoN) [inline packages/puppeteer-core/src/cdp/ChromeTargetManager.ts:362]: unrelated change: this condition is redundant because of the check on line 351 (improved for readability)

**Review by Lightning00Blade — APPROVED** [review]
